sklearn/mixture/tests/test_gmm.py

```diff
@@ -327,6 +328,32 @@ def test_1d_1component():
         assert_array_almost_equal(g.bic(X), g_full_bic)
 
 
+def assert_fit_predict_correct(model, X):
+
+    predictions_1 = model.fit(X).predict(X)
+    predictions_2 = model.fit_predict(X)
```

Comment:
I am not 100% sure what happens here, actually. (I mistrust the GMM testing). The random state seed is fixed and ``fit_predict`` uses k-means initialization with the same random state seed, so everything is deterministic and predictions_2 is independent of the previous fit, right?